src/libs/ActiveClientManager/index.js

```diff
@@ -7,13 +7,28 @@ const clientID = Str.guid();
 const maxClients = 20;
 
 let activeClients;
+
+// Whether the current clientID is set
+let didInitialize = false;
+
+let resolveReadyPromise = null;
+
+// Keeps track of the ActiveClientManager's readiness.
+const readyPromise = new Promise((res) => {
```

Comment:
NAB: Could we call `res` `resolve` here? Maybe it's obvious to most and this isn't necessary, but I think it might help newcomers (me 😅 ) follow this logic a bit better.